src/server/scripts/BrokenIsles/NeltharionsLair/boss_rokmora.cpp

```diff
@@ -0,0 +1,569 @@
+        });
+
+        delay += 1s;
+
+        _scheduler.Schedule(delay, [this](TaskContext)
+        {
+            if (Creature* ularogg = me->FindNearestCreature(NPC_ULAROGG, 50.0f))
+            {
+                ularogg->InterruptSpell(CURRENT_CHANNELED_SPELL);
+                ularogg->SetFacingTo(5.480333f);
+            }
+
+            InstanceScript* instance = me->GetInstanceScript();
+            if (!instance)
+                return;
+
+            Creature* rokmora = instance->GetCreature(DATA_ROKMORA);
+            rokmora->AI()->DoCastSelf(SPELL_BOSS_INTRO_EMERGE);
+
+            me->GetMotionMaster()->MovePoint(POINT_INTRO, NavarroggIntroPosition2);
+        });
+
+        delay += 1s;
+
+        _scheduler.Schedule(delay, [this](TaskContext)
+        {
+            if (Creature* ularogg = me->FindNearestCreature(NPC_ULAROGG, 50.0f))
+                ularogg->CastSpell(ularogg->GetPosition(), SPELL_BOSS_INTRO_CONVERSATION);
+        });
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _scheduler.Update(diff);
+    }
+
+private:
+    TaskScheduler _scheduler;
+};
+
+// 209374 - Boss 01 Intro Convo
+// Id - 1885
+class conversation_rokmora_intro : public ConversationAI
+{
+public:
+    using ConversationAI::ConversationAI;
+
+    void OnCreate(Unit* creator) override
+    {
+        Creature* navarrogg = creator->FindNearestCreatureWithOptions(50.0f, { .CreatureId = NPC_NAVARROGG, .StringId = "RokmoraIntro" });
+        if (!navarrogg)
+            return;
+
+        conversation->AddActor(CONVO_ACTOR_NAVARROGG, CONVO_ACTOR_ID_NAVARROGG, navarrogg->GetGUID());
+    }
+
+    void OnStart() override
+    {
+        if (Milliseconds const* ularoggMoveStartTime = conversation->GetLineStartTime(DEFAULT_LOCALE, CONVO_ULAROGG_INTRO_LINE_03))
+            _events.ScheduleEvent(EVENT_ROKMORA_INTRO_MOVE, *ularoggMoveStartTime + 3s);
+    }
+
+    void OnUpdate(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        switch (_events.ExecuteEvent())
+        {
+            case EVENT_ROKMORA_INTRO_MOVE:
+            {
+                Creature* ularogg = conversation->GetActorCreature(CONVO_ACTOR_ID_ULAROGG);
+                Creature* navarrogg = conversation->GetActorCreature(CONVO_ACTOR_ID_NAVARROGG);
+                if (!ularogg || !navarrogg)
+                    break;
+
+                ularogg->GetMotionMaster()->MovePath(PATH_ULAROGG, false);
+                navarrogg->GetMotionMaster()->MovePath(PATH_NAVARROGG, false);
+                break;
+            }
+            default:
+                break;
+        }
+    }
+
+private:
+    EventMap _events;
+};
+
+// 209394 - Boss 01 Intro Emerge
+class spell_rokmora_emerge : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_UPDATE_INTERACTIONS });
+    }
+
+    void HandleShatter() const
+    {
+        Creature* creatureCaster = GetCaster()->ToCreature();
```

Comment:
there's no need to cast to creature if you're not checking if it's not a creature to return